src/main/java/com/avispl/dal/avdevices/audionetworkinterface/studiotechnologies/danteleaderclock5401a/DanteLeaderClockCommunicator.java

```diff
@@ -119,82 +165,64 @@ public List<Statistics> getMultipleStatistics() throws Exception {
 	}
 
 	/**
 	 * Populate statistics for general  properties
 	 *
 	 * @param stats map of statistics
-	 * @throws Exception when fail to get general statistics
 	 */
-	private void generalProperties(Map<String, String> stats) throws Exception {
-		Document doc = Jsoup.parse(doGet(DanteLeaderClockCommands.GET_GENERAL_COMMAND.getCommand()));
-		GeneralDTO generalDTO = new GeneralDTO();
-		populateGeneralDTO(doc, generalDTO);
-		// Not populate this group if all fields are None
-		if (generalDTO.isAllNone()) {
-			throw new ResourceNotReachableException("Fail to populate statistics for General group, all properties are not in correct format.");
+	private void generalProperties(Map<String, String> stats) {
+		try {
+			Document doc = Jsoup.parse(doGet(DanteLeaderClockCommands.GET_GENERAL_COMMAND.getCommand()));
+			GeneralDTO generalDTO = new GeneralDTO();
+			populateGeneralDTO(doc, generalDTO);
+			// Not populate this group if all fields are None
+			if (generalDTO.isAllNone()) {
+				throw new ResourceNotReachableException("Fail to populate statistics for General group");
+			}
+			String groupName = DanteLeaderClockCommands.GET_GENERAL_COMMAND.getGroupName();
+			stats.put(String.format(DanteLeaderClockConstant.GROUP_PROPERTY_NAME, groupName, DanteLeaderClockMonitoringMetrics.MAIN_CLOCK_SOURCE.getPropertyName()), generalDTO.getMainClockSource());
+			stats.put(String.format(DanteLeaderClockConstant.GROUP_PROPERTY_NAME, groupName, DanteLeaderClockMonitoringMetrics.FAILOVER_CLOCK_SOURCE.getPropertyName()),
+					generalDTO.getFailOverClockSource());
+			stats.put(String.format(DanteLeaderClockConstant.GROUP_PROPERTY_NAME, groupName, DanteLeaderClockMonitoringMetrics.FORCE_PREFERRED_LEADER.getPropertyName()),
+					generalDTO.getForcePreferredLeader());
+			stats.put(String.format(DanteLeaderClockConstant.GROUP_PROPERTY_NAME, groupName, DanteLeaderClockMonitoringMetrics.CURRENT_CLOCK_SOURCE.getPropertyName()), generalDTO.getCurrentClockSource());
+			stats.put(String.format(DanteLeaderClockConstant.GROUP_PROPERTY_NAME, groupName, DanteLeaderClockMonitoringMetrics.PRIMARY_LEADER_CLOCK.getPropertyName()), generalDTO.getPrimaryLeaderClock());
+			stats.put(String.format(DanteLeaderClockConstant.GROUP_PROPERTY_NAME, groupName, DanteLeaderClockMonitoringMetrics.PRIMARY_PTPV1_STATE.getPropertyName()), generalDTO.getPrimaryPTPV1());
+			stats.put(String.format(DanteLeaderClockConstant.GROUP_PROPERTY_NAME, groupName, DanteLeaderClockMonitoringMetrics.PRIMARY_PTPV2_STATE.getPropertyName()), generalDTO.getPrimaryPTPV2());
+			stats.put(String.format(DanteLeaderClockConstant.GROUP_PROPERTY_NAME, groupName, DanteLeaderClockMonitoringMetrics.SECONDARY_PTPV1_STATE.getPropertyName()), generalDTO.getSecondaryPTPV1());
+			stats.put(String.format(DanteLeaderClockConstant.GROUP_PROPERTY_NAME, groupName, DanteLeaderClockMonitoringMetrics.SECONDARY_PTPV2_STATE.getPropertyName()), generalDTO.getSecondaryPTPV2());
+		} catch (Exception e) {
+			logger.error(e);
```

Comment:
and here :)

Reply:
Fix these error logs to contain which operation that is failed.